Build failure: XtreemFS

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix build nixpkgs#xtreemfs`

## Build log

<details>

<summary>Build Log</summary>

```
error: builder for '/nix/store/95jzyw30aix4r9lszv20drq1ig94rq8m-XtreemFS-unstable-2015-06-17.drv' failed with exit code 2;
       last 25 log lines:
       >   127 |     struct _GLIBCXX17_DEPRECATED iterator
       >       |                                  ^~~~~~~~
       > In file included from include/boost/functional/hash.hpp:6,
       >                  from include/boost/unordered/unordered_set.hpp:18,
       >                  from include/boost/unordered_set.hpp:17,
       >                  from /build/source/cpp/include/libxtreemfs/helper.h:13,
       >                  from /build/source/cpp/src/lsfs.xtreemfs/lsfs_options.cpp:14:
       > include/boost/container_hash/hash.hpp:132:33: warning: 'template<class _Arg, class _Result> struct std::unary_function' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
       > In file included from /nix/store/xzfmarrq8x8s4ivpya24rrndqsq2ndiz-gcc-13.3.0/include/c++/13.3.0/functional:49:
       > /nix/store/xzfmarrq8x8s4ivpya24rrndqsq2ndiz-gcc-13.3.0/include/c++/13.3.0/bits/stl_function.h:117:12: note: declared here
       >   117 |     struct unary_function
       >       |            ^~~~~~~~~~~~~~
       > [ 85%] Linking CXX executable lsfs.xtreemfs
       > [ 85%] Built target lsfs.xtreemfs
       > [ 87%] Building CXX object CMakeFiles/mount.xtreemfs.dir/src/fuse/fuse_adapter.cpp.o
       > In file included from /build/source/cpp/src/fuse/fuse_adapter.cpp:8:
       > /build/source/cpp/include/fuse/fuse_adapter.h:13:10: fatal error: fuse.h: No such file or directory
       >    13 | #include <fuse.h>
       >       |          ^~~~~~~~
       > compilation terminated.
       > make[3]: *** [CMakeFiles/mount.xtreemfs.dir/build.make:76: CMakeFiles/mount.xtreemfs.dir/src/fuse/fuse_adapter.cpp.o] Error 1
       > make[2]: *** [CMakeFiles/Makefile2:253: CMakeFiles/mount.xtreemfs.dir/all] Error 2
       > make[1]: *** [Makefile:91: all] Error 2
       > make[1]: Leaving directory '/build/source/cpp/build'
       > make: *** [Makefile:274: client] Error 2
       For full logs, run 'nix log /nix/store/95jzyw30aix4r9lszv20drq1ig94rq8m-XtreemFS-unstable-2015-06-17.drv'.
```

</details>

## Additional context

I think this is caused by the fuse package splitting more, need to grab the `dev` output.

This is related to ZHF: #352882

## Metadata

N/A

## Notify maintainers

(I plan on submitting a fix for this myself so I will not notify anyone yet.)

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/filesystems/xtreemfs/default.nix
+++ b/pkgs/tools/filesystems/xtreemfs/default.nix
@@ -1,20 +1,21 @@
-{ stdenv
-, lib
-, fetchFromGitHub
-, fetchpatch
-, makeWrapper
-, stripJavaArchivesHook
-, ant
-, attr
-, boost
-, cmake
-, file
-, fuse
-, jdk8
-, openssl
-, python3
-, valgrind
-, which
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  fetchpatch,
+  makeWrapper,
+  stripJavaArchivesHook,
+  ant,
+  attr,
+  boost,
+  cmake,
+  file,
+  fuse,
+  jdk8,
+  openssl,
+  python3,
+  valgrind,
+  which,
 }:
 
 stdenv.mkDerivation {
@@ -29,8 +30,15 @@ stdenv.mkDerivation {
     sha256 = "1hjmd32pla27zf98ghzz6r5ml8ry86m9dsryv1z01kxv5l95b3m0";
   };
 
-  nativeBuildInputs = [ makeWrapper python3 stripJavaArchivesHook ];
-  buildInputs = [ which attr ];
+  nativeBuildInputs = [
+    makeWrapper
+    python3
+    stripJavaArchivesHook
+  ];
+  buildInputs = [
+    which
+    attr
+  ];
 
   patches = [
     (fetchpatch {
@@ -107,7 +115,10 @@ stdenv.mkDerivation {
 
   meta = {
     description = "Distributed filesystem";
-    maintainers = with lib.maintainers; [ raskin matejc ];
+    maintainers = with lib.maintainers; [
+      raskin
+      matejc
+    ];
     platforms = lib.platforms.linux;
     license = lib.licenses.bsd3;
   };

--- a/pkgs/tools/filesystems/xtreemfs/default.nix
+++ b/pkgs/tools/filesystems/xtreemfs/default.nix
@@ -34,9 +34,9 @@ stdenv.mkDerivation {
     makeWrapper
     python3
     stripJavaArchivesHook
+    which
   ];
   buildInputs = [
-    which
     attr
   ];
 

--- a/pkgs/tools/filesystems/xtreemfs/default.nix
+++ b/pkgs/tools/filesystems/xtreemfs/default.nix
@@ -74,8 +74,8 @@ stdenv.mkDerivation {
     substituteInPlace cpp/CMakeLists.txt \
       --replace '"/lib64" "/usr/lib64"' '"${attr.out}/lib" "${lib.getLib fuse}/lib"'
 
-    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -I${fuse}/include"
-    export NIX_CFLAGS_LINK="$NIX_CFLAGS_LINK -L${fuse}/lib"
+    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -I${lib.getDev fuse}/include"
+    export NIX_CFLAGS_LINK="$NIX_CFLAGS_LINK -L${lib.getLib fuse}/lib"
 
     export DESTDIR=$out
 

